packages/runtime/container-runtime/src/opLifecycle/remoteMessageProcessor.ts

```diff
@@ -0,0 +1,94 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ISequencedDocumentMessage, MessageType } from "@fluidframework/protocol-definitions";
+import { ContainerMessageType, ContainerRuntimeMessage } from "../containerRuntime";
+import { OpDecompressor } from "./opDecompressor";
+import { OpSplitter } from "./opSplitter";
+
+export class RemoteMessageProcessor {
+    constructor(
+        private readonly opSplitter: OpSplitter,
+        private readonly opDecompressor: OpDecompressor,
+    ) { }
+
+    public get hasPartialMessages(): boolean {
+        return this.opSplitter.hasChunks;
+    }
+
+    public get partialMessages(): ReadonlyMap<string, string[]> {
+        return this.opSplitter.chunks;
+    }
+
+    public clearPartialMessagesFor(clientId: string) {
+        this.opSplitter.clearPartialChunks(clientId);
+    }
+
+    public process(remoteMessage: ISequencedDocumentMessage): ISequencedDocumentMessage {
+        const message = copy(remoteMessage);
+
+        this.opDecompressor.processMessage(message);
```

Comment:
Please add a comment to connect this code to appropriate code that does reverse transformation.
I'd actually suggest it to be a single class, that way it's easier to see dependencies, and it's more obvious what changes are required if for example one needs to add another step in processing workflow.

Reply:
Sure. This will become slightly more involved in the next PR which actually does chunking.